### PR TITLE
amd64 linux build runner

### PR DESCRIPTION
**Pull request by BruceMacD**
Add automation that automatically creates a single ollama binary for amd64 linux builds. 

Limitations:
  - Requires glibc 2.29 (the glibc version ubuntu 20.04 has packed in), ideally we build on an ubuntu 16.04 or 18.04 runner instead to maximize glibc compatibility, but that will require a custom runner. `glibc` is used by linux to access kernal functionality so it cant really be updated by an end-user without updating their OS.

Future work:
  - Ideally I'd rather just install both version of nvcc on one runner and swap between them. I tried this and I hit some issues with the wrong cuda version being referenced during builds.

### CODE REVIEW

File: .github/workflows/release-linux.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake
+
+      - name: Generate CUDA runner
+        run: |
+          go generate ./...
+
+      - name: Prepare CUDA output for upload
+        run: |
+          tar -czvf artifact-${{ matrix.cuda }}.tar.gz ./llm/llama.cpp
```
Comment (mxyng) [inline .github/workflows/release-linux.yaml:39]: Why create an archive instead of uploading the entire build directory?

Comment (BruceMacD) [reply .github/workflows/release-linux.yaml:39]: When I upload the build directory github automatically zips it, which causes the llama.cpp server executable to lose its executable bit and no longer be seen as an file which can be run.

Comment (mxyng) [reply .github/workflows/release-linux.yaml:39]: That's simple enough to fix and a lot less complex than untaring these archives

File: .github/workflows/release-linux.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          rm -rf ./artifacts/llm
+          tar -xzvf ./artifacts/artifact-12.1.0.tar.gz -C ./artifacts/
+          mkdir -p ./llm/llama.cpp/ggml/build/cuda-12/bin
+          mkdir -p ./llm/llama.cpp/gguf/build/cuda-12/bin
+          mv ./artifacts/llm/llama.cpp/ggml/build/cuda-12/bin/server ./llm/llama.cpp/ggml/build/cuda-12/bin/server
+          mv ./artifacts/llm/llama.cpp/gguf/build/cuda-12/bin/server ./llm/llama.cpp/gguf/build/cuda-12/bin/server
+          rm -rf ./artifacts/llm
+
+      - name: Build with all packaged in runners
+        run: go build -o ollama -ldflags="-w -s -X github.com/brucemacd/ollama/version.Version=${{ github.ref_name }}"
```
Comment (mxyng) [inline .github/workflows/release-linux.yaml:89]: `-ldflags` is already being set by `GOFLAGS` so this is unnecessary

```suggestion
        run: go build -o ollama
```

**Pull request closed without merging**